src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/Publishers.cs

```diff
@@ -72,13 +72,13 @@ static PublisherAddress StaticTypeRule(Type typeBeingQueried, Type configuredTyp
         }
 
         /// <summary>
-        /// Adds a dynamic rule.
+        /// Registers a rule that is invoked for each subscription to determine the address of the publisher.
         /// </summary>
-        /// <param name="dynamicRule">The rule.</param>
-        /// <param name="description">Optional description of the rule.</param>
-        public void AddDynamic(Func<Type, PublisherAddress> dynamicRule, string description = null)
+        /// <param name="rule">The function that returns the publisher's physical address for a given event type, or null if there is no publisher.</param>
+        /// <param name="description">The description of the rule.</param>
+        public void Add(Func<Type, PublisherAddress> rule, string description)
         {
-            rules.Add(new Rule(dynamicRule, description ?? "dynamic"));
+            rules.Add(new Rule(rule, description ?? "dynamic"));
```

Comment:
I think the idea was that passing in a description isn't actually optional, so let's just have something there.

However, the changes in #3918 get rid of the description entirely, so ultimately this won't matter!